### ISSUE TITLE
[BUG] Rancher provider injects whitespace into rke_config yaml which breaks the Rancher UI.

### ISSUE THREAD

**Issue by st33v** [rancher/terraform-provider-rancher2#1457]
### Rancher Server Setup
- Rancher version: v2.9.1
- Installation option (Docker install/Helm Chart): Not sure. 
- Proxy/Cert Details: n/a

### Information about the Cluster
- Kubernetes version: v1.30.6+rke2r1
- Cluster Type (Local/Downstream): Downstream
 * Infrastructure Provider = vSphere

### User Information
- What is the role of the user logged in? Admin

### Provider Information
- What is the version of the Rancher v2 Terraform Provider in use? reproduced with v5.x and v6
- What is the version of Terraform in use? Terraform v1.0.11 on windows_amd64

### Describe the bug
I'm using the rancher provider and adding a couple of add-on configuration options related to vSphere. The provider works fine, BUT when I look in the UI, I'm missing parts of my vSphere options...

<img width="728" alt="image" src="https://github.com/user-attachments/assets/f5fb0375-9c51-4427-9d3f-825b2e3ce13a" />

When I view the YAML, I see that the values ARE there, but so is a single line of whitespace (line 314):

<img width="449" alt="image" src="https://github.com/user-attachments/assets/8f91b751-6d89-4ebd-b143-f7f282b98063" />

If I remove that whitespace, the UI displays my add-on config normally.

### To Reproduce

Use the TF rpoivider to spin up a cluster with some vSphere add-on configs?

### Actual Result

The TF provider pooped a blank line into my config and broke the UI.

### Expected Result
I'd expect that the provider would clean up any undesirable characters before talking to the Rancher UI.

### Screenshots
see above.

### Additional context
nope.

**Comment by matttrach** [comment]
Have you tried chomping the value in the Terraform config?

**Comment by st33v** [comment]
Yes, I've wrapped any yaml inputs into the provider in trimspace(yamlencode(<actual_config>)). SO... I'm not even using any heredoc strings in my input.... It's all HCL that's encoded to yaml and then has trimspace() run on it. I can't sanitize my input any more than that unless you think chomp() would do something different.



________________________________

**Comment by matttrach** [comment]
We don't modify data before sending it to the API, if trimspace isn't working try using a replace like this:
`replace(yamlencode(<actual_config>), "/^[[:space:]]+$/", "")`
You can also try placing the config in a local variable and outputting it so that you can see what is generated and narrow down any problems like this.

**Comment by matttrach** [comment]
The version of Rancher mentioned in this issue is no longer supported, as such I am closing this issue, please feel free to open a new issue if this is still relevant in newer versions.
